SpriteKit-Spring.swift

```diff
@@ -244,7 +244,7 @@ public extension SKAction {
             if initialValue == nil {
 
                 initialValue = node.value(forKeyPath: keyPath) as! CGFloat
-                initialDistance = initialDistance ?? finalValue - initialValue!
+                initialDistance = initialDistance != nil ? initialDistance * initialValue - initialValue : finalValue - initialValue!
```

Comment:
Thanks for catching this issue. However I don't think this is the correct fix. It appears the behavior of the "by" param in SKAction is inconsistent:
SKAction.scaleX(by: 0.5, y: 0.5, duration:0) // will divide xScale and yScale by 2
SKAction.moveBy(x: 0.5, y: 0.5, duration: 0) // will add 0.5 to position.x and position.y

For this reason I don't think we should change the behavior of the animate(keyPath:...) functions, but just adjust the behavior of the scaleBy(...) functions above to compute the correct byValue to give to animate(keyPath:...).

Does this make sense?

Reply:
Wow you're right, great catch! I agree, since scale is the only multiplicative SKAction we should handle it as a special case in the `scaleBy` functions.